spec/reset_memo_wise_spec.rb

```diff
@@ -407,6 +491,52 @@
         let(:target) { class_with_memo }
 
         it_behaves_like "#reset_memo_wise shared examples"
+
+        context "when method name is the same as a memoized instance method" do
+          let(:class_with_memo) do
+            Class.new do
+              prepend MemoWise
+
+              def instance_one_arg_counter
+                @instance_one_arg_counter || 0
+              end
+
+              def one_arg(a) # rubocop:disable Naming/MethodParameterName
+                @instance_one_arg_counter = instance_one_arg_counter + 1
+                "instance_one_arg: a=#{a}"
+              end
+              memo_wise :one_arg
+
+              class << self
+                prepend MemoWise
+
+                def class_one_arg_counter
+                  @class_one_arg_counter || 0
+                end
+
+                def one_arg(a) # rubocop:disable Naming/MethodParameterName
+                  @class_one_arg_counter = class_one_arg_counter + 1
+                  "class_one_arg: a=#{a}"
+                end
+                memo_wise :one_arg
+              end
+            end
+          end
+
+          it "resets memoization independently" do
```

Comment:
Do you think it's overkill to also test this for the flip case - when memoization is reset on the instance method and not the class method?

Reply:
👍 done